idb/postgres/internal/writer/writer.go

```diff
@@ -139,47 +139,52 @@ func setSpecialAccounts(addresses transactions.SpecialAddresses, batch *pgx.Batc
 
 // Get the ID of the creatable referenced in the given transaction
 // (0 if not an asset or app transaction).
-func transactionAssetID(block *bookkeeping.Block, intra uint64, typeenum idb.TxnTypeEnum) uint64 {
+func transactionAssetID(txn transactions.SignedTxnWithAD, intra uint64, block *bookkeeping.Block) uint64 {
```

Comment:
Why did we need to pass `txn`? Why does the previous version not work?

Reply:
The previous version doesn't work for inner transactions because they cannot be selected from `block.Payset`. I was hoping to avoid passing in the block, but the payset length and TxnCounter are required for the old logic.

Reply:
Is this needed for a future PR? In this PR, we don't select inner txns and `intra` is not used unless `intra = idx`.

Reply:
Yes, these changes are more important for the next PR. I probably could have avoided changing this function for this PR.